### PR TITLE
T420

### PR DESCRIPTION
**Pull request by secretrobotron**
Dialog module doesn't really exist anymore, and the constituent library is much simpler.

### CODE REVIEW

**Comment by jbuck** [comment]
Some nits: 
- IFRAMEDialog looks weird, can you rename to something like InlineDialog? Looks better
- We're missing licenses on many of these files, but I think we should strip all of the licenses on individual files and just have a project-wide LICENSE file.

File: src/dialog/iframe-dialog.js [hunk 1]
```diff
@@ -27,53 +34,69 @@ define( [ "./util", "./base-dialog", "core/comm", "./event-manager" ], function(
...
+      _comm.unlisten( "cancel", onCancel );
+      _comm.unlisten( "close", _this.close );
+      _comm.destroy();
+      _open = false;
+      for( var e in _listeners ){
+        _em.unlisten( e, _listeners[ e ] );
+      } //for
       _em.dispatch( "close" );
-      _currentComm = _currentTemplate = undefined;
     } //close
```
Comment (jbuck) [inline src/dialog/iframe-dialog.js]: Needs a semi-colon

File: src/dialog/modal.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+define( [], function(){
+
+  var __container;
```
Comment (jbuck) [inline src/dialog/modal.js:3]: Rename to _container

File: src/dialog/window-dialog.js [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
-  const DEFAULT_WINDOW_HEIGHT = 479;
+  var DEFAULT_WINDOW_WIDTH = 640,
+      DEFAULT_WINDOW_HEIGHT = 479,
+      WINDOW_CHECK_INTERVAL = 300;
 
-  var WindowDialog = function( context, dialogOptions ) {
+  return function( dialogOptions ) {
     dialogOptions = dialogOptions || {};
 
     if( !dialogOptions.url ){
```
Comment (jbuck) [inline src/dialog/window-dialog.js:15]: Bah, I can't get this on the correct line, but the error message below refers to the iframe dialog, not the window dialog.

File: src/dialog/window-dialog.js [hunk 4]
```diff
@@ -33,63 +44,87 @@ define( [ "./util", "./base-dialog", "core/comm", "./event-manager" ], function(
...
+        _window.close();
+      } //if
+      clearInterval( _statusInterval );
+      _comm = _window = undefined;
+      _open = false;
+      for( var e in _listeners ){
+        _em.unlisten( e, _listeners[ e ] );
+      } //for
+      _em.dispatch( "close" );
     } //close
```
Comment (jbuck) [inline src/dialog/window-dialog.js]: Needs a ;

File: src/dialog/modal.js (hunk 2 above)
Comment (secretrobotron) [inline src/dialog/modal.js:3]: This was on purpose so that it's known that the scope is meant to be class-wide, not instance-wide.

Comment (jbuck) [inline src/dialog/modal.js:3]: I need a cheatsheet

**Pull request merged**